frontend/src/app/output/page-content/page-content.component.ts

```diff
@@ -67,9 +80,13 @@ export class PageContentComponent implements OnInit, OnDestroy {
   getScenario(part: PagePart) {
     return (part.data as ScenarioPart).scenarios;
   }
-
 }
 
 const navigateTo = (router: Router, ngZone: NgZone) => (path: string) => {
-  ngZone.run(() => router.navigateByUrl(path)).catch(err => console.log(err));
+  ngZone.run(() => {
+    const pathParams = path.split(';');
+    const slashes = pathParams[1].split('/');
+    const fragment = slashes[slashes.length - 1].split('#');
+    router.navigate([pathParams[0], {path: decodeURIComponent(slashes[0].replace('path=', ''))}, fragment[0]], {fragment: fragment[1]}).catch(err => console.log(err));
```

Comment:
Add some defensive programming.
What if path is null
What if pathParams.length == 0
?